### DISCUSSION TITLE
Enable Cors on Any Host

### DISCUSSION THREAD

**Discussion by seikosantana** [scottoffen/grapevine#62] [Q&A] [answered]
How do i enable cors from any host? I tried
`restServer.UseCorsPolicy("http://*");`
and
`restServer.UseCorsPolicy("*");`

But neither worked. Invalid URI: The hostname could not be parsed.

**Comment by scottoffen** [comment]
Wouldn't this be the same as disabling CORs? If that is your objective, just don't include the CORs middleware.

**Reply by seikosantana** [reply]
Nope, i'm trying to enable cors for any origin..

**Comment by scottoffen** [comment, marked as answer]
```csharp
sever.UseCorsPolicy();
```

This adds `OnRequest` middleware that does the following for all incoming requests:

```csharp
context.Response.AddHeader("Access-Control-Allow-Origin", "*");
context.Response.AddHeader("Access-Control-Allow-Headers", "X-Requested-With");
```

There are additional extensions that take parameters that allow you to configure the CORS policy to your liking, but this approach does the wild card CORS, which is what I think you're asking about.
